CPDShell/Core/algorithms/DensityBasedCPD/abstracts/density_based_algorithm.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Callable
+
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.neighbors import KernelDensity
+
+from CPDShell.Core.algorithms.abstract_algorithm import Algorithm
+
+
+class DensityBasedAlgorithm(Algorithm):
+    @staticmethod
+    def _kernel_density_estimation(observation: np.ndarray, bandwidth: float) -> KernelDensity:
+        """Perform kernel density estimation on the given observations.
+
+        Args:
+            observation (np.ndarray): the data points for which to estimate the density.
+            bandwidth (float): the bandwidth parameter for the kernel density estimation.
+
+        Returns:
+            KernelDensity: a fitted KernelDensity object.
+        """
+        kde = KernelDensity(kernel="gaussian", bandwidth=bandwidth).fit(observation)
+        return kde
+
+    @staticmethod
+    def _calculate_weights(
+        test_value: np.ndarray,
+        reference_value: np.ndarray,
+        bandwidth: float,
+        regularization_coef: float,
```

Comment:
It seems this argument is unused. Can it be removed or it should be used somewhere?

Reply:
Apparently I wanted to use this value to improve estimation and prevent overfitting, but decided not to implement it yet. I'll delete it for now.